src/main/scala/io/scalac/wtf/domain/UserService.scala

```diff
@@ -0,0 +1,36 @@
+package io.scalac.wtf.domain
+
+import cats.data.Validated.{Invalid, Valid}
+import cats.{Applicative, Functor, Monad}
+import cats.implicits._
+import slick.dbio.DBIO
+import slick.driver.H2Driver.api._
+
+import scala.concurrent.ExecutionContext
+
+object UserService {
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  /*
+  //TODO: Move this somewhere else
+  //Evidence for DBIO being a cats Monad
+  implicit val dbioMonad = new Functor[DBIO] with Applicative[DBIO] with Monad[DBIO] {
+    def pure[A](a: A): DBIO[A] = DBIO.successful(a)
+    def flatMap[A, B](fa: DBIO[A])(f: A => DBIO[B]) = fa.flatMap(f)
+  }
+*/
+  def createUser(createdUser: User)(implicit executionContext: ExecutionContext): DBIO[UserId] = {
+    UserRepository.findByEmail(createdUser.email).flatMap {
+      case Some(_) => DBIO.failed(new Exception("Provided email is already taken"))
```

Comment:
Ok, true. But still, we would rather gather all the validation errors here and return a combined messages to the user. I'd like it to be possible to get the return information that the password is too short and email already exists (you might want to make a test case out of this).